Sources/SmithyTestUtil/ResponseTestUtil/FloatingPointEquivalenceChecker.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public func floatingPointValuesMatch(
+    lhs: (any FloatingPoint)?,
+    rhs: (any FloatingPoint)?
+) -> Bool {
+    // Both are nil => equal
+    if (lhs == nil && rhs == nil) { return true }
+    // Only one of them is nil => not equal
+    if (lhs == nil || rhs == nil) { return false }
+    // Both are NaN => equal
+    if (lhs!.isNaN && rhs!.isNaN) { return true }
```

Comment:
I know you already verified that both lhs & rhs are non-nil, but still beware of force-unwrap.

If someone comes along & refactors or borrows this code later & isn't careful, the `!` could bite them.